### DISCUSSION TITLE
How to Get Player's Postions And How to Teleport Players to Certain Positions.

### DISCUSSION THREAD

**Discussion by TitaniumLithium** [roflmuffin/CounterStrikeSharp#481] [Q&A] [answered]
In Class `CBaseEntity` There is a method : `Teleport(Vector?, QAngle?, Vector?)`
However, Players Cannot be teleported to certain positions like this.
Note: `player` is class of `CCSPlayerController` and `Vector` is one of classes in `CounterStrikeSharp.API.Modules.Utils`
```
Vector position = new Vector(X,Y,Z);
player.Teleport(position);
```

So I have to teleport players by console commands like this:
```
var cheatsCvar = ConVar.Find("sv_cheats");
cheatsCvar.GetPrimitiveValue<bool>() = true; 
cheatsCvar.SetValue(true);
player.ExecuteClientCommandFromServer($"setpos {X} {Y} {Z}");
cheatsCvar.SetValue(false);
```
Any more effective ways?

As for getting player's positions, I did not find any method or properties.

**Comment by TitaniumLithium** [comment, marked as answer]
>  The player controller represents the player on the server, and the player pawn represents the players physical character in the game world.

Access `CCSPlayerPawn` to teleport players.
**To Get Players's Postions:**
```
CCSPlayerPawn pawn = player.PlayerPawn.Get();
Console.WriteLine(pawn.AbsOrigin);
```
This is Abs position. (different from positions in command getpos)

**To Teleport Players:**
```
CCSPlayerPawn pawn = player.PlayerPawn.Get();
Vector Position = new Vector(X,Y,Z);
pawn.Teleport(Position);
```

**Reply by YCcHxxt** [reply]
Need Help, I would like to ask someone to help me make a plugin for a transfer storage point similar to kz, email 1803035208@qq.com

**Reply by TitaniumLithium** [reply]
> Need Help, I would like to ask someone to help me make a plugin for a transfer storage point similar to kz, email [1803035208@qq.com](mailto:1803035208@qq.com)

传送除开用上面一样的方法，好像还可以用`CPointTeleport`（没研究过） 
[https://docs.cssharp.dev/api/CounterStrikeSharp.API.Core.CPointTeleport.html](https://docs.cssharp.dev/api/CounterStrikeSharp.API.Core.CPointTeleport.html) 

获取玩家当前位置坐标可以这样：
```ruby
CCSPlayerPawn pawn = player.PlayerPawn.Get();
x = pawn!.AbsOrigin!.X;
y = pawn!.AbsOrigin!.Y;
z = pawn!.AbsOrigin!.Z;
```
然后把坐标(x,y,z) 找个地方存 用的时候再拿出来 比如监听`EventPlayerSpawn`玩家重生后传送至目标点
不过kz一般是特殊地图吧 在地图做传送点不是更好
